In [2]:
import pandas as pd

df = pd.read_csv("../data/movies_raw.csv")

print(df.shape)
df.head(10)

(1336, 4)


,Film,Year,Awards,Nominations
0,One Battle After Another,2025,6,13
1,Sinners,2025,4,16
2,Frankenstein,2025,3,9
3,KPop Demon Hunters,2025,2,2
4,Hamnet,2025,1,8
5,Weapons,2025,1,1
6,Sentimental Value,2025,1,9
7,Mr Nobody Against Putin,2025,1,1
8,All the Empty Rooms,2025,1,1
9,The Singers,2025,1,1


In [3]:
print("Missing values per column:")
print(df.isnull().sum())

print("\nDuplicate rows:", df.duplicated().sum())

print("\nData types:")
print(df.dtypes)

Missing values per column:
Film           0
Year           0
Awards         0
Nominations    0
dtype: int64

Duplicate rows: 0

Data types:
Film           object
Year            int64
Awards         object
Nominations    object
dtype: object


In [4]:
df['Awards']= pd.to_numeric(df['Awards'], errors = 'coerce')
df['Nominations'] = pd.to_numeric(df['Nominations'], errors='coerce')

df = df.dropna(subset=['Awards', 'Nominations'])
df['Awards'] = df['Awards'].astype(int)
df['Nominations'] = df['Nominations'].astype(int)

print(df.shape)
print(df.dtypes)

(1279, 4)
Film           object
Year            int64
Awards          int64
Nominations     int64
dtype: object


In [5]:
df.describe().round(2)

,Year,Awards,Nominations
count,1279.00,1279.00,1279.00
mean,1978.89,1.62,3.71
std,26.77,1.39,3.20
min,1934.00,1.00,1.00
25%,1955.00,1.00,1.00
50%,1979.00,1.00,2.00
75%,2002.50,2.00,6.00
max,2025.00,11.00,16.00


In [6]:
df['Decade'] = (df['Year'] // 10) * 10

decade_counts = df['Decade'].value_counts().sort_index()
print(decade_counts)

Decade
1930     72
1940    166
1950    148
1960    142
1970    121
1980    129
1990    137
2000    147
2010    148
2020     69
Name: count, dtype: int64


In [7]:
top_awarded = df.sort_values('Awards', ascending=False).head(10)
top_awarded[['Film', 'Year', 'Awards', 'Nominations']]

,Film,Year,Awards,Nominations
878,Ben-Hur,1959,11,12
309,The Lord of the Rings: The Return of the King,2003,11,11
391,Titanic,1997,11,14
1308,West Side Story,1961,10,11
991,Gigi,1958,9,9
535,The Last Emperor,1987,9,9
402,The English Patient,1996,9,12
983,From Here to Eternity,1953,8,13
1150,On the Waterfront,1954,8,12
1128,My Fair Lady,1964,8,12


In [8]:
df['Win_Rate'] = df['Awards'] / df['Nominations']

print(df['Win_Rate'].describe())

count    1279.000000
mean        0.637488
std         0.339618
min         0.083333
25%         0.333333
50%         0.500000
75%         1.000000
max         1.000000
Name: Win_Rate, dtype: float64


In [9]:
correlation = df['Nominations'].corr(df['Awards'])
print(f"Correlation between Nominations and Awards: {correlation:.2f}")

Correlation between Nominations and Awards: 0.69


In [10]:
decade_stats = df.groupby('Decade')[['Awards', 'Nominations']].mean().round(2)
print(decade_stats)

        Awards  Nominations
Decade                     
1930      1.36         3.44
1940      1.39         3.23
1950      1.69         3.53
1960      1.70         3.99
1970      1.64         3.72
1980      1.68         3.57
1990      1.71         3.60
2000      1.63         3.76
2010      1.63         4.09
2020      1.70         4.41


In [11]:
df.to_csv("../data/movies_clean.csv", index=False)
print(f"Saved {len(df)} cleaned movies to movies_clean.csv")

Saved 1279 cleaned movies to movies_clean.csv


## EDA Findings

- Dataset: 1,279 Academy Award-winning films (1934–2025), scraped from Wikipedia
- Award-winning films peaked in the 1940s (166 films) and are least common in the 1930s and 2020s (data still incomplete for 2020s)
- Average win rate (awards won / nominations received) is 64%, with a median of 50%
- Nominations have increased over time (3.44 avg in 1930s → 4.41 avg in 2020s), while awards per film stayed relatively flat (~1.6-1.7)
- Nominations and Awards are moderately correlated (r = 0.69) — more nominations generally means more wins, but it's not a guarantee
- Most-awarded films: Ben-Hur, Titanic, and The Lord of the Rings: The Return of the King (11 awards each)